app_utils/wave_utils.py

```diff
@@ -359,3 +359,19 @@ async def report_error(q: Q):
     q.page[card_name].items[13].text_xs.visible = True
 
     await q.page.save()
+
+
+async def busy_dialog(q: Q, title: str = "", text: str = "") -> None:
+    """Creates busy dialog"""
+
+    q.page["meta"].dialog = ui.dialog(
+        title=title,
+        primary=True,
+        items=[
+            ui.progress(label=text),
+        ],
+        blocking=True,
+    )
+    await q.page.save()
+    await q.sleep(0.1)
```

Comment:
what is the sleep for?

Reply:
Replacing messages does not work in the wave chat element unfortunately.

Reply:
@pascal-pfeiffer thanks for the feedback!

About the chat, I agree that having a "..." bubble within the chat would be the best while waiting for the answer. However, to my knowledge, Wave does not support this yet and more importantly, since operations like "model.generate" are running on the main process, the app serve function gets stuck anyway. 

And for the current example, I agree it looks like chat dialog pops up only for a fraction of the  time and it can be annoying. It's also annoying that the loading wheel pops up :) I'd personally prefer seeing the dialog but this might not be the case for everyone. 

One more to add, for the screen-recording above, I used a very small backbone and submitted very short prompts. With bigger models and longer prompts, it already takes more than a couple of seconds to return the answer based on my experience (trying it on 3xRTX 6000 in our machines)

If you all think the loading wheel is better than the dialog for the chat, I can remove it 👍

Reply:
>what is the sleep for?

It's just to make sure we see something when busy_dialog is called. I actually copy pasted it from our MV repo :D

Reply:
I would stick to the loading wheel for now for the chat also.

Reply:
>I would stick to the loading wheel for now for the chat also.

Done: https://github.com/h2oai/h2o-llmstudio/pull/49/commits/e68f1444b760a917dc13a82bce58480c27701a28